Project01/Program.cs

```diff
@@ -1,2 +1,34 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿using Lesson04;
+
+Console.WriteLine("\n  Task 1");
+
+var bld = new JustBuilding();
+bld.Height = 40;
+bld.Flors = 10;
+Console.WriteLine($" высота этажа {bld.GetFlorH}");
+
+
+Console.WriteLine("\n \n   Task 2");
+int[] k = new int[3];
+k[0] = Creator.CreateBuild();
+k[1] = Creator.CreateBuild();
+k[2] = Creator.CreateBuild();
+
+foreach (int i in k)
+{
+    if (Creator.GetBuilding(i) is not null)
+        Console.WriteLine($" ключ k={i}  with building number : {Creator.GetBuilding(i).GetNumber()} ");
+
```

Comment:
```suggestion
    if (Creator.GetBuilding(i) is { } building)
        Console.WriteLine($" ключ k={i}  with building number : {building.GetNumber()} ");
```
А вот сделали бы не метод, а свойство, можно было бы написать так:

```C#
    if (Creator.GetBuilding(i) is { Number: var number })
        Console.WriteLine($" ключ k={i}  with building number : {number} ");
```